src/rust/inetstack/protocols/layer4/tcp/peer.rs

```diff
@@ -272,19 +271,10 @@ impl<N: NetworkRuntime> SharedTcpPeer<N> {
     }
 
     /// Processes an incoming TCP segment.
-    pub fn receive(&mut self, ip_hdr: Ipv4Header, buf: DemiBuffer) {
-        let (tcp_hdr, data): (TcpHeader, DemiBuffer) =
-            match TcpHeader::parse(&ip_hdr, buf, self.tcp_config.get_rx_checksum_offload()) {
-                Ok(result) => result,
-                Err(e) => {
-                    let cause: String = format!("invalid tcp header: {:?}", e);
-                    error!("receive(): {}", &cause);
-                    return;
-                },
-            };
+    pub fn receive(&mut self, src_ipv4_addr: Ipv4Addr, dest_ipv4_addr: Ipv4Addr, tcp_hdr: TcpHeader, buf: DemiBuffer) {
         debug!("TCP received {:?}", tcp_hdr);
```

Comment:
Could we take out of this PR any changes that are not related to the source-tree refactor?